x-pack/test/accessibility/apps/ml.ts

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import path from 'path';
+
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const a11y = getService('a11y');
+  const ml = getService('ml');
+
+  describe('ml', () => {
+    const esArchiver = getService('esArchiver');
+
+    before(async () => {
+      await ml.securityCommon.createMlRoles();
+      await ml.securityCommon.createMlUsers();
+    });
+
+    after(async () => {
+      await ml.securityCommon.cleanMlUsers();
+      await ml.securityCommon.cleanMlRoles();
+    });
+
+    describe('for user with full ML access', function () {
+      before(async () => {
+        await ml.securityUI.loginAsMlPowerUser();
+        await ml.api.cleanMlIndices();
+      });
+
+      after(async () => {
+        await ml.securityUI.logout();
+      });
+
+      describe('with no data loaded', function () {
+        it('overview page', async () => {
+          await ml.navigation.navigateToMl();
+          await ml.navigation.navigateToOverview();
+          await a11y.testAppSnapshot();
+        });
+
+        it('anomaly detection page', async () => {
+          await ml.navigation.navigateToAnomalyDetection();
+          await a11y.testAppSnapshot();
+        });
+
+        it('data frame analytics page', async () => {
+          await ml.navigation.navigateToDataFrameAnalytics();
+          await a11y.testAppSnapshot();
+        });
+
+        it('settings page', async () => {
+          await ml.navigation.navigateToSettings();
+          await a11y.testAppSnapshot();
+        });
+      });
+
+      describe('with data loaded', function () {
+        const adJobId = 'fq_single_permission';
```

Comment:
Nit: we might want to replace `permission` with `a11y` in all the variable values. While it doesn't impact test execution and the values could basically be anything, it reads better if they are aligned to the test suite.

Reply:
Done in https://github.com/elastic/kibana/pull/88323/commits/55ed7968b20327dd9d5205bfb6c89d877aa3fd92